core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/CoarseMesosSchedulerBackend.scala

```diff
@@ -109,6 +123,13 @@ private[spark] class CoarseMesosSchedulerBackend(
     }
   }
 
+  protected def driverUrl: String = AkkaUtils.address(
+    AkkaUtils.protocol(sc.env.actorSystem),
+    SparkEnv.driverActorSystemName,
+    conf.get("spark.driver.host"),
+    conf.get("spark.driver.port"),
+    CoarseGrainedSchedulerBackend.ACTOR_NAME)
```

Comment:
I suggested this in #4984, but this could just do a check to see if we're in a test:
```
private val driverUrl: String = {
  val testing = conf.get("spark.testing", "false").toBoolean
  if (testing) {
    "<stub"> // Mock this class without connecting to a non-existent driver in tests
  } else {
    AkkaUtils.address(...)
  }
}
```